scripts/build_scripts/build_with_profiler_opt.sh

```diff
@@ -1,4 +1,9 @@
 #! /usr/bin/env bash
+#
+# Copyright
+# © 2023 Tenstorrent Inc.
+# © 2024 Tactical Computing Labs, LLC
+#
```

Comment:
This script was deleted from main. I don't understand why I am seeing it in this diff.

Reply:
This patch was built on a specific commit that is out of sync with main; this commit point was suggested by some folks at TT. This particular script had references to Ninja.

Reply:
I've updated this branch to mirror main; will hopefully have this branch synchronized soon.

Reply:
branch should be synchronized now. I've noticed the cmake scripts require `umd::device` before it's compiled by the build script. I've build umd independent of the build script and added in `find_package` to try to detect an existing build and installation of the dependency.